x-pack/plugins/alerting/server/task_runner/task_runner.test.ts

```diff
@@ -3528,6 +3528,62 @@ describe('Task Runner', () => {
     });
   });
 
+  test('reschedules for next schedule interval if es connectivity error encountered and schedule interval is less than connectivity retry', async () => {
+    rulesClient.get.mockImplementation(() => {
+      throw SavedObjectsErrorHelpers.createGenericNotFoundEsUnavailableError('alert', '1');
+    });
+
+    const taskRunner = new TaskRunner(
+      ruleType,
+      mockedTaskInstance,
+      taskRunnerFactoryInitializerParams
+    );
+
+    encryptedSavedObjectsClient.getDecryptedAsInternalUser.mockResolvedValue({
+      id: '1',
+      type: 'alert',
+      attributes: {
+        apiKey: Buffer.from('123:abc').toString('base64'),
+        enabled: true,
+      },
+      references: [],
+    });
+
+    const runnerResult = await taskRunner.run();
+    expect(runnerResult.schedule!.interval).toEqual('10s');
```

Comment:
Just a nitpick : Maybe we can use `mockedTaskInstance.schedule?.interval` rather than a hardcoded string. iI took me some time to figure out where do we get this number from :)

Reply:
Updated in this commit: https://github.com/elastic/kibana/pull/123642/commits/1e5e9f9432d18722dbfa7a112245f2c6a19cdcc7